app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/notification/DeviceShieldNotificationScheduler.kt

```diff
@@ -185,18 +188,20 @@ class DeviceShieldDailyNotificationWorker(
     }
 
     private suspend fun showNotification() {
-        val deviceShieldNotification = deviceShieldNotificationFactory.createDailyDeviceShieldNotification().also {
-            notificationPressedHandler.notificationVariant = it.notificationVariant
-        }
+        if (ActivityCompat.checkSelfPermission(context, permission.POST_NOTIFICATIONS) == PackageManager.PERMISSION_GRANTED) {
```

Comment:
QQ: why using `ActivityCompat` instead of `notificationManagerCompat` like in other places? is there any difference?

Reply:
@aitorvs The places where I didn’t use `NotificationManagerCompat.checkPermissionAndNotify()` emits a pixel whenever the notification is shown. So when permission is not granted, not only the notification is not schedule, I also don’t send the pixel.

I can improve this either by:
1. use `NotificationManagerCompat.checkPermissionAndNotify()` and let the pixel be emitted anyway (this is the behavior today)
2. I can expose N`otificationManagerCompat.checkPermissionAndNotify(callback: () -> Unit)`

wdyt?

Reply:
Gotcha, nah, then leave it for now. We can do something later on